pkg/resource/route.go

```diff
@@ -67,9 +70,12 @@ func (gr *generatorRoute) GetName() string {
 func (gr *generatorRoute) expected() (runtime.Object, error) {
 	r := &routeapi.Route{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        gr.GetName(),
-			Namespace:   gr.GetNamespace(),
-			Annotations: map[string]string{RouteOwnerAnnotation: "true"},
+			Name:      gr.GetName(),
+			Namespace: gr.GetNamespace(),
+			Annotations: map[string]string{
+				RouteTimeoutAnnotation: "30s",
```

Comment:
given the issues we hit with this, do we perhaps want to set it even higher by default?

Reply:
also will the operator stomp this value if the user changes it?  if so it seems like we need a config api for it (though setting it explicitly to a sufficiently high value by default can go in as the immediate fix since it's better than nothing)

Reply:
After some long discussion with @dmage allied to my last findings on this issue I am inclined to just close this PR. The default timeout of 30 seconds is, as per ha-proxy documentation, for the server (in this case the image-registry) to start to reply to a request. To make this value explicit does not help, neither help to increase it as I will try to explain below.

I have done some debug around this issue and I discovered that under heavy CPU pressure image-registry may take more than 30 seconds to start the reply to a "get manifest" request (we are currently requesting for only 100m of CPU by default) and I think that a 504 response on this scenario is a valuable piece of information regarding resource exhaustion.

As per version 4.4, even with the connection drop (that happens with the 504), image registry still tries to process the request completely. This behavior has changed on 4.5 so it is important to keep this timeout lower in order to stop processing requests for which nobody is there to read the result for. IMO if we are taking more than 30 seconds to start replying one needs to scale the registry horizontally.

I would like to hear inputs on this one. I will open a separate PR to increase default memory and CPU requests, we can discuss these aspects there.

cc @dmage @adambkaplan

Reply:
your assessment makes sense to me.  if it were 30s for the total response i think we'd need something, but since it's 30s between packets(and initial packet) it doesn't seem necessary.

That said, i can envision cases where we might take 30s to do the initial response because our initial rbac lookup/layer-locating logic can be so expensive, particularly if your imagestreams are large and the apiserver is overwhelmed.  But if that's happening, the admin would, as you say, want to fix that, not raise the timeout.